sops.el

```diff
@@ -94,109 +94,95 @@
+      (kill-buffer (get-buffer "*sops-mode-process*"))
+      (erase-buffer)
+      (insert sops--original-buffer-string)
+      (save-buffer))))
 
 (defun sops-encrypt-file ()
   "Sops encrypt data."
   (run-hooks 'sops-before-encrypt-decrypt-hook)
   (if sops--sops-version-greater-than-equal-to-3-9
       (call-process-region (point-min) (point-max) sops-executable t t nil "--filename-override" sops--original-buffer-file-name "--encrypt" "/dev/stdin")
-    (progn
-      (let* ((decrypted-buffer-contents (buffer-string)))
-        (switch-to-buffer sops--original-buffer-name)
-        (erase-buffer)
-        (insert decrypted-buffer-contents)
-        (save-buffer)
-        (call-process sops-executable nil nil nil "-e" "-i" (buffer-file-name))))))
+    (let ((decrypted-buffer-contents (buffer-string)))
+      (switch-to-buffer sops--original-buffer-name)
+      (erase-buffer)
+      (insert decrypted-buffer-contents)
+      (save-buffer)
+      (call-process sops-executable nil nil nil "-e" "-i" (buffer-file-name)))))
 
 (defun sops-cancel ()
   "Cancel saving sops encrypted data."
   (interactive)
-  (if (and (bound-and-true-p sops--status) sops-mode)
-      (sops--clean-up-buffers sops--original-buffer-file-name)))
+  (when (and (bound-and-true-p sops--status) sops-mode)
+    (sops--clean-up-buffers sops--original-buffer-file-name)))
 
 (defun sops--is-sops-file ()
   "Check to see if a file is a sops encrypted file."
   (save-excursion
     (goto-char (point-min))
-    (if (search-forward "ENC[AES256_GCM" nil t 2)
-        t)))
+    (when (search-forward "ENC[AES256_GCM" nil t 2)
+      t)))
 
 (defun sops--clean-up-buffers (filename)
   "Clean up buffers we might have created using FILENAME as the suffix."
   (dolist (elt '("*sops-mode-process*-%s" "*sops-mode-process-error*-%s"))
-    (let ((formatted-elt (format elt filename)))
-      (if (get-buffer formatted-elt)
-          (kill-buffer formatted-elt)))))
+    (when-let (buffer (get-buffer (format elt filename)))
+      (kill-buffer buffer))))
 
 (defun sops--version-check ()
   "Check that sops version is greater than or equal to 3.9."
   (with-temp-buffer
-    (if (eq (call-process sops-executable nil t nil "--version") 0)
-        (progn
-          (let ((major-minor-version (s-split "\\."(cadr (s-split " " (buffer-string))))))
-            (if (and (>= (string-to-number (car major-minor-version)) 3) (>= (string-to-number (cadr major-minor-version)) 9))
-                (progn
-                  (setq sops--sops-version-greater-than-equal-to-3-9 t)
-                  t)
-              (progn
-                (setq sops--sops-version-greater-than-equal-to-3-9 nil)
-                nil)))))))
+    (when (zerop (call-process sops-executable nil t nil "--version"))
+      (goto-char (point-min))
+      (re-search-forward (rx (+ digit) (+ (and "." (+ digit)))))
+      (setq sops--sops-version-greater-than-equal-to-3-9
+            (version<= "3.9" (match-string 0))))))
```

Comment:
As I was reviewing I couldn't figure out how `(version<= "3.9" (match-string 0))` actually worked. But I would another solution I think I like better `(not (string-version-lessp (match-string 0) "3.9.0"))`

Reply:
I messed up and changed `version<=` before I looked up. Actually this is correct.

Reply:
Well, `(match-string 0)` is the sops version, so `(version<= "3.9" (match-string 0))` means the sops version `>=` (is greater than or equal to) 3.9, doesn't it? However, your suggestion is totally fine, so I will follow your preference.